help/tquery.md

```diff
@@ -1,39 +1,50 @@
 # Tquery
 
+Pole `filter`, o ile istnieje, zawiera obiekt filtra lub `always` (where true) lub `never` (where false)
+
 ### Typy danych i operatory filtrów
 
-| typ          | null(5) | = | in   | cmp | like | select | sort |
-|--------------|---------|---|------|-----|------|--------|------|
-| bool         | *       | * | .    | .   | .    | *      | +    |
-| date         | *       | * | *    | *   | .    | *      | +    |
-| datetime     | *       | . | .    | *   | .    | *      | +    |
-| int          | *       | * | *    | *   | *    | *      | +    |
-| string       | *       | * | *    | *   | *    | *      | +    |
-| uuid         | *       | * | *    | .   | .    | *      | .    |
-| text         | *       | . | .    | .   | *    | *(8)   | .    |
-| --------(9)  |         |   |      |     |      |        |      |
-| dict(1)      | *       | * | *    | .   | .    | *      | *    |
-| object(2)    | *       | . | .    | .   | .    | *      | .    |
-| dict_list(3) | *       | * | *(7) | .   | .    | *      | .    |
-| list(4)      | *       | . | .    | .   | .    | *      | .    |
-
-- `null` - operator poprawny tylko dla typów nullable
-- `=` - porównanie z wartością tego samego typu, int-int, string-string, itp., brak możliwości porównania z null ani z
-  pustym stringiem
-- `==`(?) - porównanie typu binary, bardziej kosztowne dla sql'a, ale z zachowaniem wielkości liter, polskich liter,
-  białych znaków
+| typ          | null | = | in   | cmp | like | select | sort |
+|--------------|------|---|------|-----|------|--------|------|
+| bool         | *    | * | .    | .   | .    | *      | +    |
+| date         | *    | * | *    | *   | .    | *      | +    |
+| datetime     | *    | . | .    | *   | .    | *      | +    |
+| int          | *    | * | *    | *   | *    | *      | +    |
+| string       | *    | * | *    | *   | *    | *      | +    |
+| text         | *    | . | .    | .   | *    | *(8)   | .    |
+| uuid         | *    | * | *    | .   | .    | *      | .    |
+| --------(9)  |      |   |      |     |      |        |      |
+| dict(1)      | *    | * | *    | .   | .    | *      | *    |
+| object(2)    | *    | . | .    | .   | .    | *      | .    |
+| dict_list(3) | *    | * | *(7) | .   | .    | *      | .    |
+| list(4)      | *    | . | .    | .   | .    | *      | .    |
+
+- dla wszystkich porównań `val` powinno być (zależnie od typu porównania, wartością lub listą)
+  - tego samego typu co kolumna dla `bool`, `int`, `string`, `text`-`string`
+  - stringiem poprawnie reprezentującą wartość (`date` - `Y-m-d`, `datetime` - `Y-m-d\TH:i:s\Z`, `uuid` - uuid v4)
+  - dla wszystkich operatorów dla `string` oprócz `%v`, `v%` i `%v%`, wartość musi być strimmowana
```

Comment:
Strimmowanego można oczekiwać dla =, == i in. Dla pozostałych może mieć sens. Dla like i regexp na pewno ma, ale dla porównań tekst ze spacją na końcu też jest w określonym miejscu w słowniku, w innym miejscu niż strimmowana wersja.